### DISCUSSION TITLE
Application installer

### DISCUSSION THREAD

**Discussion by MeDustyy** [AlexisL61/JackboxPatcherFR#137] [Ideas]
Create an installer for the auto-updater that installs it in Windows like any other full-fledged application (so that it can be uninstalled through the Control Panel or Settings and can be found and pinned to the Start menu).

**Comment by AlexisL61** [comment]
(Converting to discussion, that way we will have ideas in the same place)
Actually, I like to keep it portable (as you don't need to have administrator privilege to run it). On Windows 11 you can pin an executable to the start menu, that way it works fine

**Reply by MeDustyy** [reply]
You can make two versions of the application: an installer and a portable one.
I know that I can add shortcuts to any folder and to the Start Menu, but... Personally, I prefer to have the application installed on my PC.

**Reply by AlexisL61** [reply]
Ok, well that could be an idea for the future.

(Some package that could be useful for development in case this feature is built : https://pub.dev/packages/msix)

**Reply by AkiraArtuhaxis** [reply]
Sorry to interfere, but I don't agree with this change. This should be either one or the other, not both. If there's a Mac version in the future, the download page would be flooded with 6 different downloads, which is awful. Personally, I like the portable state it has currently, as it's just extract and run. Although I understand some people would prefer to have it installed so the shortcut is made automatically and such.

**Reply by MeDustyy** [reply]
> Sorry to interfere, but I don't agree with this change. This should be either one or the other, not both. If there's a Mac version in the future, the download page would be flooded with 6 different downloads, which is awful. Personally, I like the portable state it has currently, as it's just extract and run. Although I understand some people would prefer to have it installed so the shortcut is made automatically and such.

Most open source applications that I know allow you to download both a portable version and an installer. I don't see a problem with 6 download options if people need it.

**Reply by MeDustyy** [reply]
> the download page would be flooded with 6 different downloads, which is awful.

And about that, I asked for an installer only for the Auto-Updater, which has only one release. For the releases from this repository, the installer is not needed.

**Reply by AkiraArtuhaxis** [reply]
The download page would still be flooded with 6 different ways to install the Utility Updater, which is still bad. Even though there are open source apps that allow that, I prefer to keep it simple for the users. Some people nowadays don't even know what portable means.

**Reply by MeDustyy** [reply]
But most people only download the Auto-Updater once and never go back to the download page again. Is choosing one of the 6 download options going to bother people a lot?

**Comment by AkiraArtuhaxis** [comment]
Not exactly bother, more like confuse. Again, some people might not know the difference between installer and portable. The platforms will probably be clear enough to understand. But this, I don't know.

**Comment by MeDustyy** [comment]
I, on the contrary, have downloaded only installers all my life, and I feel much more comfortable when all the application files are somewhere on the disk and I have only 1 shortcut that launches the app. That way, I won't accidentally do anything with the files.

I don't think we need to continue the argument because we have opposing opinions. People can decide for themselves what is best for them. We can write a small explanation for those who don't know the difference between a portable and an installer.

**Comment by AlexisL61** [comment]
I agree that there can be a portable release and an installer release on github. In the majority, users are downloading the app from the server owner's website, so the website maker can choose to only show the portable download / installer download.

**Comment by AkiraArtuhaxis** [comment]
That's true. If the installer is straight forward, it can work. Then we can just keep one or the other depending on the owner's site.
